docs/interface/state.md

```diff
@@ -25,6 +25,52 @@ The reasoning for providing `FillDerived*` and `EstimateTimestep*` function poin
 
 In Parthenon, each `Mesh` and `MeshBlock` owns a `Packages_t` object, which is a `std::map<std::string, std::shared_ptr<StateDescriptor>>`.  The object is intended to be populated with a `StateDescriptor` object per package via an `Initialize` function as in the advection example [here](../example/advection/advection.cpp).  When Parthenon makes use of the `Packages_t` object, it iterates over all entries in the `std::map`.  Note that it's often useful to add a `StateDescriptor` to the `Packages_t` object for the overall application, allowing for a convenient way to define global parameters, for example.
 
+## History output
+
+Parthenon allows packages to enroll an arbitrary number of "history" functions that are all
+called at the interval according to the input parameters,
+see [output documention](../outputs.md#History-Files).
+
+To entroll functions create a list of callback function with the appropriate reduction operation:
+
+```c++
+// List (vector) of HistoryOutputVar that will all be enrolled as output variables
+parthenon::HstVar_list hst_vars = {};
+
+// Add a callback function
+hst_vars.emplace_back(parthenon::HistoryOutputVar({UserHistoryOperation::sum, MyHstFunction, "my label"}));
+
+// add callbacks for HST output identified by the `hist_str`
+pkg->AddParam<>(parthenon::hist_str, hst_vars);
```

Comment:
I take it the `parthenon::hist_str` is the param name and it's called in a loop over packages? I don't love the name `hist_str`. It sounds like it has a deeper meaning. Maybe `parthenon::hist_param` or `parthenon::hist_param_key` or  something?